Dockerfile

```diff
@@ -1,20 +1,22 @@
 # Build the manager binary
-FROM golang:1.13 as builder
+FROM golang:1.17 as builder
 
 WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
-COPY vendor/ vendor/
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
```

Comment:
I am not sure what your opinion about having endor here or not, so I left it as kubebuilder v3 default.

Reply:
I personally like to check in the vendor directory. But I know people do it differently and I'm fine without it.